test/hotspot/jtreg/runtime/Thread/StopAtExit.java

```diff
@@ -58,24 +57,37 @@ public void run() {
     }
 
     public static void main(String[] args) {
-        StopAtExit threads[] = new StopAtExit[N_THREADS];
+        int timeMax = 0;
+        if (args.length == 0) {
+            timeMax = DEF_TIME_MAX;
+        } else {
+            try {
+                timeMax = Integer.parseUnsignedInt(args[0]);
+            } catch (NumberFormatException nfe) {
+                System.err.println("'" + args[0] + "': invalid timeMax value.");
+                    usage();
+            }
+        }
 
-        for (int i = 0; i < N_THREADS; i++ ) {
-            threads[i] = new StopAtExit();
-            int late_count = 1;
-            threads[i].start();
+        System.out.println("About to execute for " + timeMax + " seconds.");
+
+        long count = 0;
+        long start_time = System.currentTimeMillis();
+        while (System.currentTimeMillis() < start_time + (timeMax * 1000)) {
+            count++;
+
+            StopAtExit thread = new StopAtExit();
+            thread.start();
             try {
                 // Wait for the worker thread to get going.
-                threads[i].startSyncObj.await();
-
-                // This interrupt() call will break the worker out
-                // of the exitSyncObj.await() call and the stop()
-                // calls will come in during thread exit.
-                threads[i].interrupt();
-                for (; late_count <= N_LATE_CALLS; late_count++) {
-                    threads[i].stop();
+                thread.startSyncObj.await();
+                // Tell the worker thread to race to the exit and the
+                // Thread.stop() calls will come in during thread exit.
+                thread.exitSyncObj.countDown();
+                while (true) {
+                    thread.stop();
```

Comment:
It seems the main() method of these 6 or so thread tests are identical except for the one Thread API you call at this location. Have you considered refactoring so there is only one copy of all the common code?

Reply:
Not really. It would be harder to use one of these tests in a standalone
fashion outside of JTREG runs if it depended on another .java file.